apriltag_ros/src/continuous_detector.cpp

```diff
@@ -86,6 +93,72 @@ bool ContinuousDetector::refreshParamsCallback(std_srvs::Empty::Request& req,
   return true;
 }
 
+void ContinuousDetector::publishEmptyDetection(const std_msgs::Header& header)
+{
+  AprilTagDetectionArray tag_detection_array;
+  tag_detection_array.header = header;
+  tag_detections_publisher_.publish(tag_detection_array);
+}
+
+bool ContinuousDetector::isTagInFOV(const tf::Transform& tag_pose) const
+{
+  if (!camera_model_.initialized())
+  {
+    ROS_WARN_STREAM_THROTTLE(1.0, "Camera not initialized, skipping FOV check");
+    return false;
+  }
+  if (tag_pose.getOrigin().getZ() <= 0)
+  {
+    // handles the case where Z is equal to zero (shouldn't happen) or less than zero (behind the camera)
+    return false;
+  }
+
+  // convert the 3d pose in the image frame to a 2d pixel frame
+  const auto image_pt = camera_model_.project3dToPixel(
+    cv::Point3f(
+      tag_pose.getOrigin().getX(), 
+      tag_pose.getOrigin().getY(), 
+      tag_pose.getOrigin().getZ()
+    )
+  );
```

Comment:
Maybe you could add a helper conversion function like this:
```c++
cv::Point3f toCvPoint3f(const tf::Vector3& position)
{
  return { position.getX(), position.getY(), position.getZ() };
}
```

Then you can do the following here:
```suggestion
  const auto image_pt = camera_model_.project3dToPixel(toCvPoint3f(tag_pose.getOrigin()));
```